newthread.php

```diff
@@ -510,550 +508,424 @@
+        }
+
+        // One or more errors returned, fetch error list and throw to newreply page
+        if (count($post_errors) > 0) {
+            $thread_errors = inline_error($post_errors);
+        } else {
+            $query = $db->query("
+                SELECT u.*, f.*
+                FROM ".TABLE_PREFIX."users u
+                LEFT JOIN ".TABLE_PREFIX."userfields f ON (f.ufid=u.uid)
+                WHERE u.uid='".$mybb->user['uid']."'
+            ");
+            $post = $db->fetch_array($query);
+            $post['username'] = $username;
+            if ($mybb->user['uid']) {
+                $post['userusername'] = $mybb->user['username'];
+            }
+
+            $previewmessage = $mybb->get_input('message');
+            $post['message'] = $previewmessage;
+            $post['subject'] = $mybb->get_input('subject');
+            $post['icon'] = $mybb->get_input('icon', MyBB::INPUT_INT);
+
+            $mybb->input['postoptions'] = $mybb->get_input('postoptions', MyBB::INPUT_ARRAY);
+            if (isset($mybb->input['postoptions']['disablesmilies'])) {
+                $post['smilieoff'] = $mybb->input['postoptions']['disablesmilies'];
+            }
+
+            $post['dateline'] = TIME_NOW;
+            if (isset($mybb->input['postoptions']['signature'])) {
+                $post['includesig'] = $mybb->input['postoptions']['signature'];
+            }
+
+            if (!isset($post['includesig']) || $post['includesig'] != 1) {
+                $post['includesig'] = 0;
+            }
+
+            // Fetch attachments assigned to this post
+            if ($mybb->get_input('pid', MyBB::INPUT_INT)) {
+                $attachwhere = "pid='".$mybb->get_input('pid', MyBB::INPUT_INT)."'";
+            } else {
+                $attachwhere = "posthash='".$db->escape_string($mybb->get_input('posthash'))."'";
+            }
+
+            $query = $db->simple_select("attachments", "*", $attachwhere);
+            while ($attachment = $db->fetch_array($query)) {
+                $attachcache[0][$attachment['aid']] = $attachment;
+            }
+
+            $newthread['preview'] = true;
+            $postbit = build_postbit($post, 1);
+        }
+
+        $newthread['message'] = $mybb->get_input('message');
+        $newthread['subject'] = $mybb->get_input('subject');
+    }
+
+    // Removing an attachment or adding a new one, or showing thread errors.
+    else if ($mybb->get_input('attachmentaid', MyBB::INPUT_INT) || $mybb->get_input('newattachment') || $mybb->get_input('updateattachment') || $thread_errors) {
+        $newthread['message'] = $mybb->get_input('message');
+        $newthread['subject'] = $mybb->get_input('subject');
+    }
+
+    // Generate thread prefix selector
+    if (!$mybb->get_input('threadprefix', MyBB::INPUT_INT)) {
+        $mybb->input['threadprefix'] = 0;
+    }
+
+    $prefixselect = build_prefix_select($forum['fid'], $mybb->get_input('threadprefix', MyBB::INPUT_INT));
```

Comment:
As I commented in the other PR, it might make sense to create a new version of this function that just returns an array.